contracts/src/core/EigenDAThresholdRegistry.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import {EigenDAThresholdRegistryStorage} from "./EigenDAThresholdRegistryStorage.sol";
+import {IEigenDAThresholdRegistry} from "../interfaces/IEigenDAThresholdRegistry.sol";
+import {OwnableUpgradeable} from "@openzeppelin-upgrades/contracts/access/OwnableUpgradeable.sol";
+import {BitmapUtils} from "eigenlayer-middleware/libraries/BitmapUtils.sol";
+import "../interfaces/IEigenDAStructs.sol";
+
+contract EigenDAThresholdRegistry is EigenDAThresholdRegistryStorage, OwnableUpgradeable {
+
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize(
+        address _initialOwner,
+        bytes memory _quorumAdversaryThresholdPercentages,
+        bytes memory _quorumConfirmationThresholdPercentages,
+        bytes memory _quorumNumbersRequired,
+        uint16[] memory _versions,
+        VersionedBlobParams[] memory _versionedBlobParams,
+        SecurityThresholds memory _defaultSecurityThresholdsV2
+    ) external initializer {
+        _transferOwnership(_initialOwner);
+
+        quorumAdversaryThresholdPercentages = _quorumAdversaryThresholdPercentages;
+        quorumConfirmationThresholdPercentages = _quorumConfirmationThresholdPercentages;
+        quorumNumbersRequired = _quorumNumbersRequired;
+        defaultSecurityThresholdsV2 = _defaultSecurityThresholdsV2;
+        
+        require(_versions.length == _versionedBlobParams.length, "EigenDAThresholdRegistry: versions and versioned blob params length mismatch");
+        for (uint256 i = 0; i < _versions.length; ++i) {
+            versionedBlobParams[_versions[i]] = _versionedBlobParams[i];
+        }
+    }
+
+    function updateQuorumAdversaryThresholdPercentages(bytes memory _quorumAdversaryThresholdPercentages) external onlyOwner {
+        quorumAdversaryThresholdPercentages = _quorumAdversaryThresholdPercentages;
+    }
+
+    function updateQuorumConfirmationThresholdPercentages(bytes memory _quorumConfirmationThresholdPercentages) external onlyOwner {
+        quorumConfirmationThresholdPercentages = _quorumConfirmationThresholdPercentages;
+    }
+
+    function updateQuorumNumbersRequired(bytes memory _quorumNumbersRequired) external onlyOwner {
+        quorumNumbersRequired = _quorumNumbersRequired;
+    }
+
+    function updateVersionedBlobParams(uint16 version, VersionedBlobParams memory _versionedBlobParams) external onlyOwner {
```

Comment:
What are your thoughts on making these values immutable and append only? 
Instead of allowing caller to set values for arbitrary versions, can we make it append only and enforce you can only add a new version in increments of 1?